ppb_vector/vector2.py

```diff
@@ -56,35 +57,37 @@ def _find_lowest_vector(left: typing.Type, right: typing.Type) -> typing.Type:
         return _find_lowest_type(left, right)
 
 
+@dataclass(eq=False, frozen=True, init=False, repr=False)
```

Comment:
This should probably have `eq=True` for reasons of hashing?

Reply:
That should result in a `TypeError` since `Vector2` defines it's own `__eq__`.

Moreover, I don't think we can make it an hashable type, as AFAIK `hash(vector) == hash(vector_like)` would need to hold wherever `vector == vector_like`.

Reply:
Yes, for this you want to key the same, two equal things must hash the same.

In this case, I don't think we want to key the same to anything not an actual vector? And this becomes a problem not because we expect users to key dictionaries on vectors, but because of things like `functools.lru_cache`, which use dictionaries under the hood.